security-analysis/security-analysis-api/src/main/java/com/powsybl/security/SecurityAnalysisProvider.java

```diff
@@ -130,4 +135,43 @@ default CompletableFuture<SecurityAnalysisReport> run(Network network,
                                                           List<StateMonitor> monitors) {
         return run(network, workingVariantId, detector, filter, computationManager, parameters, contingenciesProvider, interceptors);
     }
+
+    /**
+     * Reads implementation-specific parameters from platform config, or return {@link Optional#empty()}
+     * if the implementation does not have any specific parameters, or does not support loading from config.
+     *
+     * @return The specific parameters read from platform config.
+     */
+    default Optional<Extension<LoadFlowParameters>> loadSpecificParameters(PlatformConfig config) {
+        return Optional.empty();
+    }
+
+    /**
+     * Reads implementation-specific parameters from a Map, or return {@link Optional#empty()}
+     * if the implementation does not have any specific parameters, or does not support loading from config.
+     *
+     * @return The specific parameters read from Map.
+     */
+    default Optional<Extension<LoadFlowParameters>> loadSpecificParameters(Map<String, String> properties) {
+        return Optional.empty();
+    }
+
+    /**
+     * Reads implementation-specific parameters from a Map, or return {@link Optional#empty()}
+     * if the implementation does not have any specific parameters, or does not support loading from config.
+     *
+     * @return The specific parameters read from Map.
+     */
+    default Optional<Extension<LoadFlowParameters>> loadLoadflowSpecificParameters(Map<String, String> properties) {
```

Comment:
As just discussed, after more thoughts I think it's better to have a link to the LF provider via its name:
```java
Optional<String> getLoadFlowName();
```

This way we can have access to all methods corresponding to the parameters of this loadflow, including the parameters names.

If a security analysis does not have a correspond LF implementation, it does not make sense to have specific LF parameters for that security analysis, so it should be OK.

@geofjamg : does it sound OK for you ?

Reply:
Yes agree!